code/ui/blocks/src/blocks/ArgTypes.stories.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import type { Meta, StoryObj } from '@storybook/react';
+
+import { ArgTypes } from './ArgTypes';
+import * as ButtonStories from '../examples/Button.stories';
+
+const meta: Meta<typeof ArgTypes> = {
+  title: 'Blocks/ArgTypes',
+  component: ArgTypes,
+  parameters: {
+    relativeCsfPaths: ['../examples/Button.stories', '../blocks/ArgTypes.stories'],
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default: Story = {};
+
+export const OfComponent: Story = {
+  args: {
+    of: ButtonStories.default.component,
+  },
+};
+
+export const OfMeta: Story = {
+  args: {
+    of: ButtonStories.default,
+  },
+};
+
+export const OfStory: Story = {
+  args: {
+    of: ButtonStories.Primary,
+  },
+};
+
+// NOTE: this will throw with no of prop
+export const OfStoryUnattached: Story = {
+  parameters: { attached: false },
+  args: {
+    of: ButtonStories.Primary,
+  },
+};
+
+export const Simple = {
+  render: () => <div>Story for reference</div>,
+  argTypes: {
+    a: { type: { name: 'string' }, name: 'A', description: 'a' },
+    b: { type: { name: 'string', required: true }, name: 'B', description: 'b' },
+  },
+};
+
+export const IncludeProp: Story = {
+  args: {
+    of: Simple,
+    include: ['A'],
+  },
+};
+
+export const SimpleInclude = {
+  ...Simple,
+  parameters: { docs: { argTypes: { include: ['A'] } } },
+};
+
+export const IncludeParameter: Story = {
+  args: {
+    of: SimpleInclude,
+  },
+};
+
+export const ExcludeProp: Story = {
```

Comment:
This story looks a little confusing to me, since the table shows args for the `ArgTypes`  block which is very meta. Could we reference Example Button instead?
![image](https://user-images.githubusercontent.com/5678122/213111382-a3516f9d-1e12-4429-b74d-d66f923d61f9.png)

Reply:
sounds fine by me. we might add comments to them describing what they're used for and where.

Reply:
Ok, done in https://github.com/storybookjs/storybook/pull/20664/commits/c6fd9b959a7e8db96dc275367fe0aaa529d26c1c. I think it's much nicer actually. I'll update source + story to do the same in another PR.